saleor/graphql/order/tests/test_order.py

```diff
@@ -6524,3 +6524,96 @@ def test_get_variant_from_order_line_variant_not_exists_as_staff(
     content = get_graphql_content(response)
     orders = content["data"]["me"]["orders"]["edges"]
     assert orders[0]["node"]["lines"][0]["variant"] is None
+
+
+QUERY_ORDER_BY_TOKEN_WITH_PAYMENT = """
+    query OrderByToken($token: UUID!) {
+        orderByToken(token: $token){
+            id
+            payments{
+              id
+              gateway
+              isActive
+              created
+              modified
+              paymentMethodType
+              transactions{
+                gatewayResponse
+
+              }
+              actions
+              availableCaptureAmount{
+                amount
+              }
+              availableRefundAmount{
+                amount
+              }
+
+              creditCard{
+                brand
+                firstDigits
+                lastDigits
+                expMonth
+                expYear
+              }
+            }
+        }
+  }
+"""
+
+QUERY_ORDER_WITH_PAYMENT_AVAILABLE_FIELDS = """
+    query OrderByToken($token: UUID!) {
+        orderByToken(token: $token){
+            id
+            payments{
+              id
+              gateway
+              isActive
+              created
+              modified
+              paymentMethodType
+              availableCaptureAmount{
+                amount
+              }
+            chargeStatus
+              creditCard{
+                brand
+                firstDigits
+                lastDigits
+                expMonth
+                expYear
+              }
+            }
+        }
+  }
+"""
+
+
+def test_order_by_token_query_for_payment_details_without_permissions(
+    api_client, payment_txn_captured
+):
+    response = api_client.post_graphql(
+        QUERY_ORDER_BY_TOKEN_WITH_PAYMENT, {"token": payment_txn_captured.order.token}
+    )
+    assert_no_permission(response)
+
+    response = api_client.post_graphql(
+        QUERY_ORDER_WITH_PAYMENT_AVAILABLE_FIELDS,
+        {"token": payment_txn_captured.order.token},
+    )
+
+    content = get_graphql_content(response)
+    assert "errors" not in content, content
```

Comment:
`assert "errors" not in content, content` --> This is already checked in `get_graphql_content` method.



Also, you should split it into two separate tests. For the second test, I suggest checking order `id` to check if you get the correct `order`.